tests/gradients/test_finite_difference.py

```diff
@@ -392,7 +395,7 @@ class DeviceSupportingSpecialObservable(DefaultQubit):
 
             @staticmethod
             def _asarray(arr, dtype=None):
-                return arr
+                return np.asarray(arr)

```

Comment:
Pennylane's, ie. Autograd's, NumPy arrays behave _much_ more reasonable when they are zero-dimensional, therefore a custom implementation should use those, not vanilla numpy arrays.
An example is that `np.array(MyCoolObject(0.4), dtype=object) * 0.3` returns an array with Pennylane's NumPy, but a `MyCoolObject` instance for vanilla NumPy (provided that `__mul__` is implemented and returns a `MyCoolObject` instance itself).